hack/update-codegen.sh

```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-ANTREA_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+ANTREA_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..
```

Comment:
why is this change made?

Reply:
Command ''$(dirname "${BASH_SOURCE[0]}")/..'' can only get a relative path. For example
``` bash
 ~/go/src/github.com/vmware-tanzu/antrea cat ./hack/update-codegen.sh
#!/usr/bin/env bash
ANTREA_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
echo $ANTREA_ROOT
 ~/go/src/github.com/vmware-tanzu/antrea# ./hack/update-codegen.sh
./hack/..
```
Command ''$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..'' can get an absolute path.
```bash
 ~/go/src/github.com/vmware-tanzu/antrea# cat ./hack/update-codegen.sh
#!/usr/bin/env bash
ANTREA_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..
IMAGE_NAME="antrea/codegen:kubernetes-1.18.4"

echo $ANTREA_ROOT%
 ~/go/src/github.com/vmware-tanzu/antrea# ./hack/update-codegen.sh
/Users/lhongliang/go/src/github.com/vmware-tanzu/antrea/hack/..
```
Here is the full scripts, as the folder `hack` will be mounted to a container, $ANTREA_ROOT should be an absolute path.